client/client.go

```diff
@@ -139,6 +143,59 @@ func (c *Client) BuildNonJSONRequest(method, path string, body io.ReadSeeker) (*
 	return req, nil
 }
 
+// Builds a new HTTP Request for a multipart form request
+func (c *Client) BuildMultipartFormRequest(method, path string, files map[string]string, parameters map[string]interface{}) (*http.Request, error) {
+	urlPath, err := url.Parse(path)
+	if err != nil {
+		return nil, err
+	}
+
+	body := new(bytes.Buffer)
+	writer := multipart.NewWriter(body)
+
+	for fileName, filePath := range files {
+		// Open the file
+		file, err := os.Open(filePath)
+		if err != nil {
+			return nil, err
+		}
+
+		// Read the file contents
+		fileContents, err := ioutil.ReadAll(file)
+		if err != nil {
+			return nil, err
+		}
+
+		// Write out the file information and contents
+		fi, err := file.Stat()
+		if err != nil {
+			return nil, err
+		}
+		file.Close()
```

Comment:
use `defer` under the `Open` err check or close it in all the `err` cases (just use `defer`)